src/frontend/window/main_window.rs

```diff
@@ -43,17 +49,31 @@ impl AppModel {
         }
     }
 
+    fn notify(&self, msg: String, urgency: Urgency) {
+        let handle = std::thread::spawn(move || {
+            if let Err(err) = Notification::new()
+                .appname(crate_name!())
+                .urgency(urgency)
+                .summary(FLAKESHOT_SUMMARY)
+                .body(&msg)
+                .show()
+            {
+                error!("Couldn't show notification: {}", err);
+            }
+        });
```

Comment:
Why are you creating a new thread and WAIT for it to finish? Why not just executing the code on the main thread?

```suggestion
 if let Err(err) = Notification::new()
            .appname(crate_name!())
            .urgency(urgency)
            .summary(FLAKESHOT_SUMMARY)
            .body(&msg)
            .show()
        {
            error!("Couldn't show notification: {}", err);
        }
```

Reply:
This won't work for two reasons:

1. `show()` will create a new async-runtime which will crash
2. We'll have a data-race with the auto-exit feature for the clipboard: Flakeshot isn't fast enough to send to notification to the notification server.

Reply:
1. It doesn't crash here. Why should it?
2. Doesn't happen here neither. Anyway the `save_to_clipboard` should not send the quit command. You can add it to `handle_tool_event` and there set the right order.

Reply:
> It doesn't crash here. Why should it?

because you can't invoke two nested tokio-async-runtimes

> Doesn't happen here neither. Anyway the save_to_clipboard should not send the quit command. You can add it to handle_tool_event and there set the right order.

well it does while I tried it

Reply:
welp, sorry about the confusion